### DISCUSSION TITLE
`complete_shell` doesn't seem to work as expected in bash

### DISCUSSION THREAD

**Discussion by teohhanhui** [pacak/bpaf#366] [Q&A]
If I use

```rust
        .complete_shell(ShellComp::File { mask: None })
```

on this line: https://github.com/teohhanhui/hyfetch/blob/b955aa23167eaaccd1a1ac1e5c2d2ec8053f7778/crates/hyfetch/src/cli_options.rs#L35

it insists on showing all the files in the current working directory, regardless of the current input before `TAB`.

Using

```rust
        .complete_shell(ShellComp::Nothing)
```

together with

```bash
_bpaf_dynamic_completion()
{
    line="$1 --bpaf-complete-rev=8 ${COMP_WORDS[@]:1}"
    if [[ ${COMP_WORDS[-1]} == "" ]]; then
        line="${line} \"\""
    fi
    source <( eval ${line})
}
complete -o nosort -o bashdefault -o default -F _bpaf_dynamic_completion hyfetch
```

in `~/.local/share/bash-completion/completions/hyfetch` seems to work well.

**Comment by pacak** [comment]
Looks like a bug, we'll see what I can do about it.

**Reply by teohhanhui** [reply]
Also, I'm wondering if `complete_shell` supports the use case of non-CWD? i.e. `hyfetch --config-file ~/.config/[TAB]` or `hyfetch --config-file /etc/[TAB]`

**Reply by pacak** [reply]
It should I think. As far as file completion is concerned it tries to offload as much work as possible to bash completers, code that consumes `ShellComp` looks like this and writes are going directly to bash:

```rust
    for op in ops {
        match op {
            ShellComp::File { mask: None } => write!(res, "_filedir"),
            ShellComp::File { mask: Some(mask) } => {
                writeln!(res, "_filedir '{}'", Shell(bashmask(mask)))
            }
            Shell Comp::Dir { mask: None } => write!(res, "_filedir -d"),
            ShellComp::Dir { mask: Some(mask) } => {
                writeln!(res, "_filedir -d '{}'", Shell(bashmask(mask)))
            }
            ShellComp::Raw { bash, .. } => writeln!(res, "{}", Shell(bash)),
            ShellComp::Nothing => Ok(()),
        }?;
    }
```
